pkg/clair/client.go

```diff
@@ -3,29 +3,51 @@ package clair
 import (
 	"bytes"
 	"crypto/tls"
+	"database/sql"
 	"encoding/json"
 	"fmt"
-	"github.com/goharbor/harbor-scanner-clair/pkg/etc"
 	"io/ioutil"
+	"log"
 	"net/http"
+	"strconv"
 	"strings"
+	"time"
+
+	"github.com/goharbor/harbor-scanner-clair/pkg/etc"
+	"github.com/xo/dburl"
+)
+
+const (
+	updaterLast = "updater/last"
 )
 
 // Client communicates with clair endpoint to scan image and get detailed scan result
 type Client interface {
 	ScanLayer(layer Layer) error
 	GetLayer(layerName string) (*LayerEnvelope, error)
+	GetVulnerabilityDatabaseUpdatedAt() (*time.Time, error)
 }
 
 type client struct {
+	db          *sql.DB
 	endpointURL string
 	// need to customize the logger to write output to job log.
 	client *http.Client
 }
 
 // NewClient constructs a new client for Clair REST API pointing to the specified endpoint URL.
 func NewClient(tlsConfig etc.TLSConfig, cfg etc.ClairConfig) Client {
+	var db *sql.DB
+	if cfg.DatabaseURL != "" {
```

Comment:
I don't think we should return an error here as CVE timestamp is an optional feature. Logging it is ok.

Reply:
this feature enabled only clair database url configed, return error is ok when users enable this feature but adapter can not connect the db